File: app/src/main/java/bassamalim/hidaya/adapters/FilteredRecyclerAdapter.java
package bassamalim.hidaya.adapters;

import androidx.recyclerview.widget.RecyclerView;

public abstract class FilteredRecyclerAdapter<VH extends RecyclerView.ViewHolder>
        extends RecyclerView.Adapter<VH> {

    public abstract void filter(String text, boolean[] selected);

}

File: app/src/main/java/bassamalim/hidaya/replacements/FilteredRecyclerAdapter.kt
package bassamalim.hidaya.replacements

import androidx.recyclerview.widget.RecyclerView

abstract class FilteredRecyclerAdapter<VH : RecyclerView.ViewHolder?> : RecyclerView.Adapter<VH>() {

    abstract fun filter(text: String?, selected: BooleanArray?)

}